File: services/api/app/fulfillment/incoterms.py
INCOTERMS = {
    # Seller/Buyer responsibility is a simple hint for UI; not a legal definition.
    "EXW": {
        "label": "Ex Works",
        "required_fields": ["origin_city", "origin_country", "pickup_ready_date"],
        "buyer_responsibilities": ["Export clearance", "Main carriage", "Insurance", "Import clearance", "Delivery to destination"],
        "seller_responsibilities": ["Goods ready at premises"],
        "notes": "Buyer handles all logistics from seller's door."
    },
    "FCA": {
        "label": "Free Carrier",
        "required_fields": ["origin_city", "origin_country", "handoff_location"],
        "buyer_responsibilities": ["Main carriage", "Insurance", "Import clearance", "Delivery"],
        "seller_responsibilities": ["Export clearance", "Handoff to carrier at named place"]
    },
    "FAS": {"label":"Free Alongside Ship","required_fields":["origin_port","cutoff_date"]},
    "FOB": {"label":"Free On Board","required_fields":["origin_port","vessel_cutoff_date"]},
    "CFR": {"label":"Cost & Freight","required_fields":["origin_port","destination_port"]},
    "CIF": {"label":"Cost, Insurance & Freight","required_fields":["origin_port","destination_port","insured_value_usd"]},
    "CPT": {"label":"Carriage Paid To","required_fields":["handoff_location","destination_city","destination_country"]},
    "CIP": {"label":"Carriage & Insurance Paid","required_fields":["handoff_location","destination_city","destination_country","insured_value_usd"]},
    "DAP": {"label":"Delivered At Place","required_fields":["destination_city","destination_country","delivery_address"]},
    "DPU": {"label":"Delivered at Place Unloaded","required_fields":["destination_city","destination_country","unload_site"]},
    "DDP": {
        "label":"Delivered Duty Paid",
        "required_fields":["origin_city","origin_country","destination_city","destination_country","delivery_address"],
        "notes":"Seller handles all logistics & duties to destination."
    }
}

File: apps/web/src/constants/incoterms.ts
export const INCOTERMS = [
  "EXW","FCA","FAS","FOB","CFR","CIF","CPT","CIP","DAP","DPU","DDP"
] as const;

export const INCOTERM_FIELD_HINTS: Record<string, {required: string[]; optional: string[]; tip: string;}> = {
  EXW: { required:["origin_city"], optional:["origin_port","dest_city","dest_port"], tip:"Buyer picks up at seller's site; buyer handles export" },
  FCA: { required:["origin_city"], optional:["origin_port","dest_city","dest_port"], tip:"Seller hands to buyer's carrier; seller clears export" },
  FAS: { required:["origin_port"], optional:["dest_port"], tip:"Alongside vessel at origin port" },
  FOB: { required:["origin_port"], optional:["dest_port"], tip:"Onboard vessel at origin port" },
  CFR: { required:["origin_port","dest_port"], optional:[], tip:"Cost+Freight to destination port; risk onboard" },
  CIF: { required:["origin_port","dest_port"], optional:[], tip:"Like CFR + seller insurance" },
  CPT: { required:["dest_city"], optional:["origin_city"], tip:"Carriage paid to named place" },
  CIP: { required:["dest_city"], optional:["origin_city"], tip:"Like CPT + insurance" },
  DAP: { required:["dest_city"], optional:["origin_city"], tip:"Delivered at place (not unloaded); buyer imports" },
  DPU: { required:["dest_city"], optional:["origin_city"], tip:"Delivered unloaded; buyer imports" },
  DDP: { required:["dest_city"], optional:["origin_city"], tip:"Seller handles import+duties to destination" },
};